src/transport_ofi.c

```diff
@@ -479,31 +515,35 @@ static inline struct fi_info * query_for_fabric(char *provname)
 
     /* find fabric provider to use that is able to support RMA and ATOMICS */
     ret = fi_getinfo( FI_VERSION(OFI_MAJOR_VERSION, OFI_MINOR_VERSION),
-		      NULL, NULL, 0, &hints, &p_info);
+		      NULL, NULL, 0, &hints, p_info);
     if(ret!=0){
-	RAISE_ERROR(ret);
+	OFI_ABORT("getinfo didn't find any providers\n");
+	return ret;
     }
 
-    if (!p_info) {
-	    OFI_ABORT("fi_getinfo didn't find any providers\n");
+    if(*p_info && (*p_info)->tx_attr->inject_size >= sizeof(long double)) {
```

Comment:
Is this assuming that long double is the longest type used by SHMEM?  I'm not sure if the C standard allows double complex to be longer than long double?